django_select2/static/django_select2/django_select2.js

```diff
@@ -23,12 +23,19 @@
             page: params.page,
             field_id: $element.data('field_id')
           }
-
           var dependentFields = $element.data('select2-dependent-fields')
           if (dependentFields) {
             dependentFields = dependentFields.trim().split(/\s+/)
             $.each(dependentFields, function (i, dependentField) {
-              result[dependentField] = $('[name=' + dependentField + ']', $element.closest('form')).val()
+              var formValue = $('[name=' + dependentField + ']', $element.closest('form')).val();
+              // This is for inlines, I checked this for a specific case
+              if (formValue === null || formValue === undefined) {
+                  var newFieldName = $element[0].name.split('-', 2).join('-') + '-' + dependentField;
+                  var formsetValue = $('[name=' + newFieldName + ']', $element.closest('.form-row')).val();
+                  result[dependentField] = formsetValue;
+              } else {
+                  result[dependentField] = formValue;
+              }
```

Comment:
I believe you can simply this a little and drop the else branch, like so:

```suggestion
              var formValue = $('[name=' + dependentField + ']', $element.closest('form')).val();
              // This is for inlines, I checked this for a specific case
              if (formValue === null || formValue === undefined) {
                  var newFieldName = $element[0].name.split('-', 2).join('-') + '-' + dependentField;
                  formValue = $('[name=' + newFieldName + ']', $element.closest('.form-row')).val();
              }
              result[dependentField] = formValue;
```